File: server/php/db/MongodbModel.php
<?php

class MongodbModel
{
    static private $conn = null;
    private $mongodb = null;

    private function __construct($db)
    {
        $conn = new MongoClient();
        $this->mongodb = $conn->$db;
        return $conn->$db;
    }
    /**
     * 初始化
     */
    static public function init($db)
    {
        if (!isset(self::$conn)) {
            self::$conn = new self($db);
        }
        return self::$conn;
    }
    /**
     * 获取所有数据
     */
    public function getAllData($table)
    {
        $res = $this->mongodb->$table->find();
        $data = array();
        foreach ($res as $k => $v) {
            $data[$k] = $v;
            $data[$k]['id']=(string) $v['_id'];//类型强制转换
        }
        return $data;
    }
    /**
     * 获取一条数据
     */
    public function getOneData($table, $where = array())
    {
        $where_q=array(
            '_id'=>new MongoId($where['id'])
        );
        $res = $this->mongodb->$table->find($where_q);
        $data = array();
        foreach ($res as $k => $v) {
            $data[$k] = $v;
            $data[$k]['id']=(string) $v['_id'];//类型强制转换
        }
        return $data;
    }
    /**
     * 添加数据
     */
    public function add($table, $data)
    {
        $ins_data = array(
            "title" => $data['title'],
            "categories" => $data['categories'],
            "keywords" => $data['keywords'],
            "content" => $data['content']
        );

        $res = $this->mongodb->$table->insert($ins_data);
        //返回执行结果
        if ($res) {
            return 1;
        } else {
            return 0;
        }
    }
    /**
     * 删除数据
     */
    public function del($table, $where = array())
    {
        $where_d=array(
            '_id'=>new MongoId($where['id'])
        );
        $res = $this->mongodb->$table->remove($where_d, array('justOne' => true));

        //返回执行结果
        if ($res) {
            return 1;
        } else {
            return 0;
        }
    }
    /**
     * 修改数据
     */
    public function edit($table, $where = array(), $data = array())
    {
        $where_e=array(
            '_id'=>new MongoId($where['id'])
        );
        $res = $this->mongodb->$table->update($where_e, array('$set' => $data));
        
        //返回执行结果
        if ($res) {
            return 1;
        } else {
            return 0;
        }
    }

    /**
     * 防止用户复制对象
     */
    private function __clone()
    {

    }

}

File: server/python/db/mongodbModel.py
import pymongo
from bson.objectid import ObjectId


class MongodbModel:
    # mongodb数据库
    def __init__(self):
        # 连接数据库
        _client = pymongo.MongoClient("mongodb://localhost:27017/")
        # 使用数据库
        self._db = _client['article']

    def getAllData(self, table):
        # 使用数据集
        _model = self._db[table]
        # 返回所有数据
        res = _model.find()
        # 定义数组
        arr = []
        for v in res:
            arr.append({
                "id": str(v['_id']),
                "title": v['title'],
                "categories": v['categories'],
                "keywords": v['keywords'],
                "content": v['content']
            })
        return arr

    def getOneData(self, table, where):
        # 使用数据集
        _model = self._db[table]
        # 构建查询条件
        where_q = {'_id': ObjectId(where['id'])}
        # 返回结果
        res = _model.find_one(where_q)
        # 定义数组
        arr = []
        if res:
            arr.append({
                "id": str(res['_id']),
                "title": res['title'],
                "categories": res['categories'],
                "keywords": res['keywords'],
                "content": res['content']
            })
            return arr

    def add(self, table, data):
        # 使用数据集
        _model = self._db[table]
        # 插入数据
        res = _model.insert_one(data)
        # 返回文档id
        insid = res.inserted_id
        if insid:
            return 1
        else:
            return 0

    def edit(self, table, where, data):
        # 使用数据集
        _model = self._db[table]
        # 查询条件
        where_e = {'_id': ObjectId(where['id'])}
        # 修改的数据
        updata = {"$set": data}
        # 执行修改
        res = _model.update_one(where_e, updata)
        # 返回执行结果
        return res.modified_count

    def delete(self, table, where):
        # 使用数据集
        _model = self._db[table]
        # 条件
        where_d = {'_id': ObjectId(where['id'])}
        # 执行删除
        res = _model.delete_one(where_d)
        # 返回结果
        return res.deleted_count